template/build/webpack.dev.conf.js

```diff
@@ -29,6 +29,14 @@ const devWebpackConfig = merge(baseWebpackConfig, {
     quiet: true, // necessary for FriendlyErrorsPlugin
     watchOptions: {
       poll: config.dev.poll,
+    },
+    before: function(app) {
+        // Redirect to index.html in assetsPublicPath if path is not set to root
+        if (config.dev.assetsPublicPath && config.dev.assetsPublicPath != '/') {
```

Comment:
1. I think `!==` is better, stylistically.
2. Just thinking: Could this break anything for a relative path ('./')?

Reply:
this has to be tested before merge.